Refactoring testing

mattgrill opened the pull request:
![](https://media.giphy.com/media/26FxykLpuDxQiFbnG/giphy.gif)
- `npm i` to update dependencies. 
- `npm t` to check tests.
- `npm run coverage` to check coverage.

Moves away from a tightly coupled relationship between entities and entity handlers. A single generic entity handler accepts parameters and configures what it needs. Tests are also much faster, 💯

wimleers commented on lib/entity.js:
```diff
@@ -0,0 +1,65 @@
+const Request = require('./helpers/request');
+const methods = require('./helpers/methods');
+
+module.exports = class Entity extends Request {
```
`Entity` sounds like the correct name to me :)

mattgrill replied:
@wimleers Thanks. At some point I will probably roll the `entity query` work into this, so we don't have two separate ways to get items.

wimleers commented on lib/hydrant.js:
```diff
@@ -14,10 +11,10 @@ module.exports = class Hydrant {
+      content: new Entity(base, credentials, {get: 'node', set: 'node', create: 'entity/node', delete: 'node'}),
+      contentType: new Entity(base, credentials, {get: '/entity/node_type', set: '/entity/node_type', create: '/entity/node_type', delete: '/entity/node_type'}),
+      menu: new Entity(base, credentials, {get: 'entity/menu', set: 'entity/menu', create: 'entity/menu', delete: 'entity/menu'}),
+      comment: new Entity(base, credentials, {get: 'comment', set: 'comment', create: 'comment', delete: 'comment'}),
```
lgtm

Commit:
Refactor entity structure, updates to testing.

Updates testing to use AVA/NYC. Add .editorconfig and change package name
to prepare for publish on NPM.

Commit:
Adding Travis

Commit:
Adding Slack integration

Commit:
Add helper functions for resources

prestonso commented on lib/hydrant.js:
```diff
@@ -13,13 +10,54 @@ module.exports = class Hydrant {
+      menu: new Entity(base, credentials, {get: 'entity/menu', set: 'entity/menu', create: 'entity/menu', delete: 'entity/menu'}),
+      taxonomyTerm: new Entity(base, credentials, {get: 'taxonomy/term', set: 'taxonomy/term', create: 'taxonomy/term', delete: 'taxonomy/term'}),
+      taxonomyVocabulary: new Entity(base, credentials, {get: 'entity/taxonomy_vocabulary', set: 'entity/taxonomy_vocabulary', create: 'entity/taxonomy_vocabulary', delete: 'entity/taxonomy_vocabulary'}),
+      user: new Entity(base, credentials, {get: 'user', set: 'user', create: 'user', delete: 'user'}),
```
w00t

prestonso commented on test/entityQuery.js:
```diff
@@ -0,0 +1,126 @@
+
+});
+
+test.cb('Contion', t => {
```
`'Condition'`

prestonso commented:
Nice work! Just had one small nit.

Commit:
Spelling

prestonso commented:
![elijah-wood-i-love-this](https://cloud.githubusercontent.com/assets/484252/15826219/bb62924e-2bd4-11e6-9eb6-d85c02da6609.gif)

lgtm :shipit:
